docker/Dockerfile.test

```
@@ -12,13 +12,14 @@ ENV SHORT_SHA $SHORT_SHA
 
 ENV RUST_BACKTRACE full
 ENV CARGO_HOME /zebra/.cargo/
+ENV RUSTFLAGS -g
```

Can just add the optimisation flag in this PR?
And remove all the other changes, so we go back to using the default dev profile?

When we activate the release profile, Cargo also turns off debug assertions and integer overflow checks. But these checks help our tests find bugs.

It might also be helpful to add a comment explaining why we optimise these builds.

```suggestion
ENV RUSTFLAGS -O
```